packages/schema/src/config/common.ts

```diff
@@ -158,7 +158,7 @@ export default defineUntypedSchema({
     $resolve: async (val: string | undefined, get): Promise<string> => {
       const isV4 = ((await get('future') as Record<string, unknown>).compatibilityVersion === 4)
 
-      return resolve(await get('rootDir') as string, (val || isV4) ? 'server' : resolve(await get('srcDir') as string, 'server'))
+      return val ? val : resolve(await get('rootDir') as string, isV4 ? 'server' : resolve(await get('srcDir') as string, 'server'))
```

Comment:
We should still resolve val if it is provided.

Reply:
sure !